integration_tests/tests/transformation-tests.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  putObjectToS3,
+  checkIfS3ObjectExists,
+  S3Object,
+  getS3ItemsList,
+  s3GetObjectsToArray,
+  deleteS3FolderBasedOnDate,
+} from "../helpers/s3Helper";
+import { resourcePrefix } from "../helpers/envHelper";
+import path from "path";
+import fs from "fs";
+import csv from "csvtojson";
+import { waitForTrue } from "../helpers/commonHelpers";
+import {
+  CsvRow,
+  TransformationEventBodyObject,
+} from "../../src/handlers/transformation/transform-row";
+
+const prefix = resourcePrefix();
+const storageBucket = `${resourcePrefix()}-storage`;
+const folderPrefix = "btm_transactions";
+
+describe("\n Upload verify fake csv to transformation bucket tests and check valid events from csv match with s3 events", () => {
+  beforeAll(async () => {
+    // before the test runs deletes the btm_transaction/datePrefix folder created by previous run
+    await deleteS3FolderBasedOnDate(storageBucket, folderPrefix);
+    console.log("Existing sub folders within btm_transactions are deleted");
+    const fakeCsvFile = "../payloads/fakeBillingReport.csv";
+    const filePath = path.join(__dirname, fakeCsvFile);
+    const csvFileData = fs.readFileSync(filePath);
+    const testObject: S3Object = {
+      bucket: `${prefix}-transformation`,
+      key: `fakeBillingReport.csv`,
+    };
+    await putObjectToS3(testObject, csvFileData);
+    const checkCsvFileExists = await checkIfS3ObjectExists(testObject);
+    expect(checkCsvFileExists).toBeTruthy();
+  });
+
+  test("events which are satisfied client and event name matching rule from csv should be stored in s3 storage bucket", async () => {
+    const checkEventsExistsInS3 = await waitForTrue(
+      checkExpectedEventsExistsInS3,
+      1000,
+      10000
+    );
+    expect(checkEventsExistsInS3).toBe(true);
+    const eventsFromS3 = await getEventsFromS3();
+    const eventsFromCSV = await filterPermittedClientAndBillingEvents();
+    const s3ContainsEventsFromCSV = eventsFromS3.some((s3Obj) => {
```

Comment:
Should this be eventsFromS3.every? Because every event in the storage/btm_transactions/{dates} folder should match the filtered events from the csv file, right?

Reply:
As discussed on Zoom, need to relook at checkExpectedEventsExistsInS3() and getEventsFromS3(). Ideally, the length of getEventsFromS3() and filterPermittedClientAndBillingEvents() is the same.

Reply:
added condition to check length of s3 equals to length of filtered csv events